django/contrib/humanize/templatetags/humanize.py

```diff
@@ -80,6 +80,8 @@ def intcomma(value, use_l10n=True):
     if match:
         prefix = match[0]
         prefix_with_commas = re.sub(r"\d{3}", r"\g<0>,", prefix[::-1])[::-1]
+        # Remove a leading comma, if needed.
+        prefix_with_commas = re.sub(r"^(-?),", r"\1", prefix_with_commas)
```

Comment:
Or if we want to keep it in the regex, a negative lookahead for the eol? 🤔 
```suggestion
        prefix_with_commas = re.sub(r"\d{3}(?!$)", r"\g<0>,", prefix[::-1])[::-1]
```

This gives me:

```
>>> re.sub(r"\d{3}(?!$)", r"\g<0>,", "333123"[::-1])[::-1]
333,123
```

Reply:
but again, it won't work for negative values:
```
>>> re.sub(r"\d{3}(?!$)", r"\g<0>,", "-333123"[::-1])[::-1]   
'-,333,123'
```

Reply:
@felixxm sorry I edited my comment to at `-?` to the negative lookahead 😅

Reply:
Thanks, TIL.

Reply:
It seems that negative lookaheads are quite inefficient, I think it's safer to restore the previous version. We don't want to re-introduce ReDoS here. Thanks David for a suggestion.